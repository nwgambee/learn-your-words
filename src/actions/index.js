export const getVocabWord = query => ({
    type: 'GET_WORD',
    query
})