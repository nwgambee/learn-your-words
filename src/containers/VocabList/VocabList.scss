.list-container {
  height: 100rem;
  background-color: #882FF6;
  display: flex;
  align-content: flex-start;
  justify-content: space-around;
  flex-direction: row;
  flex-wrap: wrap;
}

.empty-message {
  color: white;
  font-size: 2rem;
}

.hidden {
  display: none;
}

.list-card {
  margin-top: 2rem;
  margin: 2rem;
  padding: 1rem;
  padding-top: 0;
  border-radius: 2rem;
  width: 30rem;
  height: 35rem;
  background-color: blanchedalmond;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  box-shadow: 8px 8px 8px black;
}

.list-word-h1 {
  text-align: center;
  font-size: 4rem;
  margin: 0;
  border-bottom: 4px solid black;
  padding-bottom: .3rem;
}

.list-definition {
  color: blueviolet;
  text-align: center;
  font-size: 2rem;
  margin-top: 3rem  ;
}

.list-part-of {
  text-align: center;
  font-size: 2rem;
  margin: 0;
}

.remove-from-list-btn {
  margin-top: 1rem;
  background-color: #F99D07;
  padding: .75rem;
  font-size: 1.5rem;
  width: 20rem;
  border: 5px solid #35D461;
  border-radius: .5rem;
  cursor: pointer;
  font-weight: 700;
  align-self: center;
}