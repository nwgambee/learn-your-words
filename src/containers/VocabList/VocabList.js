import React, { Component } from 'react';
import { connect } from 'react-redux';
import './VocabList.scss'
import { removeFromList } from '../../actions/index'
import PropTypes from 'prop-types';


export class VocabList extends Component {
  constructor() {
    super();
 }

 render() {
   let listCards = this.props.vocabList.map(word => {
     return (
            <section className='list-card' key={word.word}>
              <h1 className='list-word-h1'>{word.word}</h1>
              <h2 className='list-definition'>{word.results[0].definition}</h2>
              <h2 className='list-part-of'>{word.results[0].partOfSpeech}</h2>
              <button className='remove-from-list-btn' onClick={() => this.props.removeFromList(word.word)}>Remove From List</button>
            </section> 

   )
  })
   return (
    <section className='list-container'>
      {listCards}
    </section>
   )
 }
}

export const mapStateToProps = state => ({
  vocabList: state.vocabList
})

export const mapDispatchToProps = dispatch => ({
  removeFromList: (id) => dispatch(removeFromList(id))
})



export default connect(mapStateToProps, mapDispatchToProps)(VocabList);

VocabList.propTypes = {
  vocabList: PropTypes.array,
  removeFromList: PropTypes.func
}
