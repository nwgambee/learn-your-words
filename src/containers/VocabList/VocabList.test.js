import React from 'react';
import { removeFromList } from '../../actions/index' 
import { VocabList, mapStateToProps, mapDispatchToProps } from './VocabList';
import { shallow } from 'enzyme';

describe('VocabList', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<VocabList vocabList=
      {
        [
          {word: 'word', results: [
            {definition: 'this is a def', partOfSpeech: 'noun'},
             {}
            ]
          }
        ]
      }
    />);
    expect(wrapper).toMatchSnapshot();
  })
  describe('mapStateToProps', () => {
    it('should return an object with the vocabList array', () => {
      const mockState = {
        searchedWord: {word: 'hat', definition: 'this is a definition', type: 'noun'},
        loadingStatus: true,
        vocabList: [{word: 'hat'}, {word: 'cat'}],
        someOtherProp: false
      }
      const expected = {
        vocabList: [{word: 'hat'}, {word: 'cat'}],
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
  
  describe('mapDispatchToProps', () => {
    it('should call dispatch with a removeFromList action when removeFromList is called', () => {
      const mockDispatch = jest.fn();
      const action = removeFromList('word');
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.removeFromList('word');
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })
  })
})
