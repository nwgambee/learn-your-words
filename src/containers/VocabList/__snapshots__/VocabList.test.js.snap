// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VocabList should match the snapshot 1`] = `
<section
  className="list-container"
>
  <section
    className="list-card"
  >
    <h1
      className="list-word-h1"
    >
      word
    </h1>
    <h2
      className="list-definition"
    >
      this is a def
    </h2>
    <h2
      className="list-part-of"
    >
      noun
    </h2>
    <button
      className="remove-from-list-btn"
      onClick={[Function]}
    >
      Remove From List
    </button>
  </section>
</section>
`;
